server/resp/src/main/java/org/infinispan/server/resp/serialization/lua/LuaResponseWriter.java

```diff
@@ -0,0 +1,285 @@
+package org.infinispan.server.resp.serialization.lua;
+
+import static org.infinispan.server.resp.scripting.LuaContext.filterCause;
+import static org.infinispan.server.resp.scripting.LuaContext.luaPushError;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+import org.infinispan.commons.jdkspecific.CallerId;
+import org.infinispan.server.resp.RespVersion;
+import org.infinispan.server.resp.serialization.JavaObjectSerializer;
+import org.infinispan.server.resp.serialization.Resp3Type;
+import org.infinispan.server.resp.serialization.ResponseWriter;
+import org.infinispan.server.resp.serialization.SerializationHint;
+
+import party.iroiro.luajava.Lua;
+
+public class LuaResponseWriter implements ResponseWriter {
+   private final Lua lua;
+   private RespVersion version = RespVersion.RESP3; // TODO: once we support RESP2, this should be lowered
+
+   public LuaResponseWriter(Lua lua) {
+      this.lua = lua;
+   }
+
+   @Override
+   public RespVersion version() {
+      return version;
+   }
+
+   @Override
+   public void version(RespVersion version) {
+      this.version = version;
+   }
+
+   @Override
+   public boolean isInternal() {
+      return true;
+   }
+
+   @Override
+   public void nulls() {
+      lua.checkStack(1);
+      lua.pushNil();
+   }
+
+   @Override
+   public void ok() {
+      lua.checkStack(1);
+      lua.push("OK");
+   }
+
+   @Override
+   public void queued(Object ignore) {
+      throw new UnsupportedOperationException("queued(Object)");
+   }
+
+   @Override
+   public void simpleString(CharSequence value) {
+      lua.checkStack(1);
+      if (value == null) {
+         lua.pushNil();
+      } else {
+         lua.push(value.toString());
+      }
+   }
+
+   @Override
+   public void string(CharSequence value) {
+      lua.checkStack(1);
+      if (value == null) {
+         lua.pushNil();
+      } else {
+         lua.push(value.toString());
+      }
+   }
+
+   @Override
+   public void string(byte[] value) {
+      lua.checkStack(1);
+      if (value == null) {
+         lua.pushNil();
+      } else {
+         lua.push(new String(value, StandardCharsets.US_ASCII));
```

Comment:
I don't know if we can assume US_ASCII here. I believe this can contain any user value, so it can be any byte value.

Reply:
I've changed it to use ISO8859-1 instead and added an explanation